packages/react/src/Footnotes/Footnotes.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import {ReplyIcon} from '@primer/octicons-react'
+import clsx from 'clsx'
+
+import {Heading, InlineLink, Text} from '../'
+import type {BaseProps} from '../component-helpers'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/footnotes/colors-with-modes.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './Footnotes.module.css'
+
+export const FootnotesTags = ['div', 'ol'] as const
+
+type FootnotesRestrictedPolymorphism =
+  | (React.HTMLAttributes<HTMLDivElement> & BaseProps<HTMLDivElement> & {as?: 'div'})
+  | (React.HTMLAttributes<HTMLParagraphElement> & BaseProps<HTMLOListElement> & {as?: 'ol'})
+
+type FootnotesProps = {
+  as?: (typeof FootnotesTags)[number]
+  visuallyHiddenHeading?: string
+} & FootnotesRestrictedPolymorphism
+
+function Root({as = 'ol', className, children, visuallyHiddenHeading = 'Footnotes', ...rest}: FootnotesProps) {
+  if (as === 'div') {
+    const filteredChildren = React.Children.toArray(children).map(child => {
+      if (React.isValidElement(child) && typeof child.type !== 'string') {
+        if (child.type === Item) {
+          return React.cloneElement(child, {
+            variant: 'disclaimer',
+            ...child.props,
+          })
+        }
+      }
+    })
+    return (
+      <>
+        <Heading className="visually-hidden" as="h2" id="footnote-label">
+          {visuallyHiddenHeading}
+        </Heading>
+        <div
+          className={clsx(styles.Footnotes, styles['Footnotes--variant-disclaimer'], className)}
+          {...(rest as BaseProps<HTMLDivElement>)}
+        >
+          {filteredChildren}
+        </div>
+      </>
+    )
+  }
+
+  return (
+    <>
+      <Heading className="visually-hidden" as="h2" id="footnote-label">
+        {visuallyHiddenHeading}
+      </Heading>
+      <ol
+        className={clsx(styles.Footnotes, styles['Footnotes--variant-citations'], className)}
+        {...(rest as BaseProps<HTMLOListElement>)}
+      >
+        {children}
+      </ol>
+    </>
+  )
+}
+
+export const FootnotesItemTags = ['a', 'p'] as const
+
+type DefaultFootnotesItemProps = {
+  variant?: 'disclaimer'
+}
+
+type CitationFootnotesItemProps = {
+  variant?: 'citation'
+  href?: string
+}
+
+type FootnotesItemProps = (DefaultFootnotesItemProps | CitationFootnotesItemProps) &
+  React.HTMLAttributes<HTMLParagraphElement> &
+  BaseProps<HTMLParagraphElement>
```

Comment:
Same here, don't think we need `BaseProps`

Reply:
Responded [here](https://github.com/primer/brand/pull/707#discussion_r1727166349)